laspy/base.py

```diff
@@ -124,11 +129,7 @@ def __init__(self, filename, manager):
 
     def open(self, mode):
         '''Open the file, catch simple problems.'''
-        if (not self.compressed) or self.mode == "r-":
-            try:
-                self.fileref = open(self.filename, mode)
-            except(Exception):
-                raise laspy.util.LaspyException("Error opening file. Do you have the right permissions?")
+        self.fileref = open(self.filename, mode)
```

Comment:
laspy has a very bad habit of catching exceptions for you and swallowing the useful information that might be in them. This block adds no value and takes away from the user's ability to figure out why their file won't open.